src/displayapp/screens/settings/SettingSetDate.cpp

```diff
@@ -0,0 +1,272 @@
+    lv_obj_align(lblYear, lv_scr_act(), LV_ALIGN_CENTER, POS_X_YEAR, POS_Y_TEXT);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+    CheckDay();
+  }
+
+  if(object == btnYearMinus && (event == LV_EVENT_PRESSED)) {
+    yearValue--;
+    lv_label_set_text_fmt(lblYear, "%d", yearValue);
+    lv_obj_align(lblYear, lv_scr_act(), LV_ALIGN_CENTER, POS_X_YEAR, POS_Y_TEXT);
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_RELEASED);
+    CheckDay();
+  }
+
+  if(object == btnSetTime && (event == LV_EVENT_PRESSED)) {
+    NRF_LOG_INFO("Setting date (manually) to %04d-%02d-%02d", yearValue, monthValue, dayValue);
+    dateTimeController.SetTime(static_cast<uint16_t>(yearValue),
+                               static_cast<uint8_t>(monthValue),
+                               static_cast<uint8_t>(dayValue),
+                               0,
+                               dateTimeController.Hours(),
+                               dateTimeController.Minutes(),
+                               dateTimeController.Seconds(),
+                               nrf_rtc_counter_get(portNRF_RTC_REG));
+    lv_btn_set_state(btnSetTime, LV_BTN_STATE_DISABLED);
+  }
+}
+
+int SettingSetDate::MaximumDayOfMonth() const {
+  switch (monthValue) {
+    case 2:
+      if ((((yearValue % 4) == 0) && ((yearValue % 100) != 0)) || ((yearValue % 400) == 0))
+        return 29;
+      return 28;
+    case 4:
+    case 6:
+    case 9:
+    case 11:
+      return 30;
+    default:
+      return 31;
+  }
+}
+
+void SettingSetDate::CheckDay() {
+  int maxDay = MaximumDayOfMonth();
+  if (dayValue > maxDay) {
+    dayValue = maxDay;
+    lv_label_set_text_fmt(lblDay, "%d", dayValue);
+    lv_obj_align(lblDay, lv_scr_act(), LV_ALIGN_CENTER, POS_X_DAY, POS_Y_TEXT);
+  }
+}
+
+void SettingSetDate::UpdateMonthLabel() {
+  switch (monthValue) {
+    case 1:
+      lv_label_set_text_static(lblMonth, "Jan");
+      break;
+    case 2:
+      lv_label_set_text_static(lblMonth, "Feb");
+      break;
+    case 3:
+      lv_label_set_text_static(lblMonth, "Mar");
+      break;
+    case 4:
+      lv_label_set_text_static(lblMonth, "Apr");
+      break;
+    case 5:
+      lv_label_set_text_static(lblMonth, "May");
+      break;
+    case 6:
+      lv_label_set_text_static(lblMonth, "Jun");
+      break;
+    case 7:
+      lv_label_set_text_static(lblMonth, "Jul");
+      break;
+    case 8:
+      lv_label_set_text_static(lblMonth, "Aug");
+      break;
+    case 9:
+      lv_label_set_text_static(lblMonth, "Sep");
+      break;
+    case 10:
+      lv_label_set_text_static(lblMonth, "Oct");
+      break;
+    case 11:
+      lv_label_set_text_static(lblMonth, "Nov");
+      break;
+    case 12:
+      lv_label_set_text_static(lblMonth, "Dec");
+      break;
+    default:
+      lv_label_set_text_static(lblMonth, "---");
+      break;
+  }
+  lv_label_set_align(lblMonth, LV_LABEL_ALIGN_CENTER);
+  lv_obj_align(lblMonth, lv_scr_act(), LV_ALIGN_CENTER, POS_X_MONTH, POS_Y_TEXT);
+}
```

Comment:
This already exists in https://github.com/JF002/InfiniTime/blob/61a4642221fc9fcab6889221e7bf9c29778589f2/src/components/datetime/DateTimeController.cpp#L119
Can you use that instead?

Reply:
Done. I had to add some public static methods (overloaded variants of the existing methods) to `DateTime` to make the arrays accessible in the screens.